### ISSUE TITLE
adding a new subserver reset config of the parent

### ISSUE THREAD

**Issue by kintaro1981** [Real-Gecko/virtualmin-nginx-fpm#19]
I think that this issue is connects to:

https://github.com/Real-Gecko/virtualmin-nginx-fpm/issues/17
https://github.com/Real-Gecko/virtualmin-nginx-fpm/issues/15

**Comment by Real-Gecko** [comment]
Yeah, man. I definitely need to rework config file creation :)

**Comment by ThomasFreedman** [comment]
Virtualmin doesn't support php-fpm for GPL release. Your work here is all I've seen that does.

I see 2 files in the install folder here, one appears to be a debian package. However both files are 2 years old and I see issues here much newer than that. I'm guessing you haven't updated the code since you posted on the Virtualmin forum in 2015 that you were using this plugin in production.

Am I correct that both files install the same plugin, one thru W/Vmin the other via OS?

**Comment by kintaro1981** [comment]
@ThomasFreedman virtualmin implemented php-fpm in the latests releases 
take a look here:
https://github.com/virtualmin/virtualmin-nginx/blob/master/CHANGELOG

**Comment by ThomasFreedman** [comment]
Yes, but in the GPL release? I've been installing for a week and have not seen it in the UI, tho I'm not looking for alternative downloads (just installing based on tutorials and most reference the same URL). Version I'm using is GPL 5.07.

**Comment by Real-Gecko** [comment]
https://github.com/Real-Gecko/virtualmin-nginx-fpm/issues/11#issuecomment-282572743

**Comment by 7starsone** [comment]
Hello, on CentOS ... is enough to do `yum install php-fpm` ? or...?

**Comment by ThomasFreedman** [comment]
I looked at comment 11 you linked to but didn't see that it addressed if native fpm support is available in GPL release.

Also, on the instructions you don't say if the native nginx module is necessary. I think I may have erroneously posted elsewhere your plugin is all you need, but I don't think you support DNS, and I believe the native nginx plugin / module does. I may be completely wrong on both counts, but what I DO know is that I installed a new VPS with vituralmin native nginx support and your plugin, unchecked the boxes for nginx website and nginx ssl website and there are no DNS entries created for new virtual servers. After reviewing your instructions you didn't say to unckeck either of native nginx-website features, but if left checked a run time error msg occurs, so they are mutually exclusive features.

If one of these nginx plugins / modules are responsible for updating DNS records there seems to be a problem, as that isn't happening. There may be some other bug in virtualmin, as the bind option is unchecked on features & plugins and if I try to enable it an error msg says I need to add 127.0.0.1 to list of DNS servers. Clicking the link in the error msg allows me to do so, but doesn't appear to "take" as the bind feature remains unchecked and can't be enabled (circular issue; error resets option, dns list can't be updated if option is not set?).

It is probably obvious by now I'm a relative noob on Web / Virtualmin. There seem to be many subtleties in the order and sequence of doing things. I got the DNS working, still don't know if your plugin negates the need for the native nginx-website & nginx-website-ssl plugins or not. The DNS problem I mentioned here might have been as easy to resolve as making sure the 127.0.0.1 entry in the DNS servers list is first, or perhaps not using the features and plugins error msg link to correct the list. I used webmin--> servers-->Bind DNS Server to add 127.0.0.1 to list and then enabled the BIND feature under virtualmin features and plugins which stuck. THEN I had to enable the DNS option on each virtual server and THAT populated the DNS entries and restarted the DNS server.

**Comment by kintaro1981** [comment]
@ThomasFreedman with the core plugin:
Server Configuration => Website Options => Switch from FCGId to FPM

**Comment by ThomasFreedman** [comment]
I see no such option there.
![screenshot from 2017-04-13 22 48 39](https://cloud.githubusercontent.com/assets/11077042/25032653/904d6432-209b-11e7-98aa-dec493176ed2.png)

**Comment by kintaro1981** [comment]
What version of virtualmin are you running?

(As we are talking about the official plugin maybe it's better to move this question on the virtualmin issue tracker on virtualmin.com)

**Comment by ThomasFreedman** [comment]
GPL 5.07, which I posted above. 

Frankly I don't find the forum all that helpful. I've never gotten a reply there and there is no sorting of search results or way to list by date. Lots of old stuff that is obsolete. Can waste a lot of time trying to find relevant info that up to date.
